Nos/Models/Event+CoreDataClass.swift

```diff
@@ -239,35 +302,43 @@ public class Event: NosManagedObject {
             signature: signature
         )
     }
-	
+
     // swiftlint:disable function_body_length
-	convenience init(context: NSManagedObjectContext, jsonEvent: JSONEvent) throws {
-		self.init(context: context)
-		
-		// Meta data
-		createdAt = Date(timeIntervalSince1970: TimeInterval(jsonEvent.createdAt))
-		content = jsonEvent.content
-		identifier = jsonEvent.id
-		kind = jsonEvent.kind
-		signature = jsonEvent.signature
+    convenience init(context: NSManagedObjectContext, jsonEvent: JSONEvent) throws {
+        self.init(context: context)
+        identifier = jsonEvent.id
+        try hydrate(from: jsonEvent, in: context)
+    }
+        
+    /// Populates an event stub (with only its ID set) using the data in the given JSON.
+    func hydrate(from jsonEvent: JSONEvent, in context: NSManagedObjectContext) throws {
+        guard isStub else {
```

Comment:
@cjorgensen @shanecb @jcswift23 I wanted to call attention to this because it is a change in the data model that I don't want to trip anyone up. I introduced the concept of a "stub" Event which is basically just an event with an `identifier` and nothing else. I'm creating these stubbed events when I see an EventReferences for an event we don't have in our database. It's important to do this so that if we do eventually download the event it has the proper relationship to the original even that referenced it.

So PSA when dealing with events that sometimes you might see events with only an identifier and no other properties set. If you are fetching with a filter on `kind` you shouldn't need to worry as the `kind` won't be set on stubs so they won't be fetched from Core Data in most cases.